p2p/src/authenticated/discovery/actors/peer/actor.rs

```diff
@@ -70,12 +70,16 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Rng + CryptoRng + Metrics, C: Pub
                 sent_messages: cfg.sent_messages,
                 received_messages: cfg.received_messages,
                 rate_limited: cfg.rate_limited,
-                reservation,
+                _reservation: reservation,
             },
             Relay::new(low_sender, high_sender),
         )
     }
 
+    pub(crate) fn mailbox(&self) -> &Mailbox<C> {
```

Comment:
Ah, this is annoying. Didn't think of this.

We may want to just return a clone of the sender with relay (think that'll be cleaner than this function although don't love that either).

Reply:
Personally, I'd be OK leaving this as is. Your call.

Reply:
Still thinking about it.

Reply:
For consistency with other parts of the code, I recommend we return the mailbox from `new`. There is nothing wrong with this reference approach but everywhere else we need to use a mailbox we return it exclusively in `new`.